tool/tsh/mfa.go

```diff
@@ -524,6 +524,13 @@ func promptWebauthnRegisterChallenge(ctx context.Context, origin string, cc *wan
 func promptTouchIDRegisterChallenge(origin string, cc *wanlib.CredentialCreation) (*proto.MFARegisterResponse, registerCallback, error) {
 	log.Debugf("Touch ID: prompting registration with origin %q", origin)
 
+	if origin == "" {
+		return nil, nil, trace.BadParameter("origin required")
+	}
+	if err := cc.Validate(true /* alwaysCreateRK */); err != nil {
+		return nil, nil, trace.Wrap(err)
+	}
```

Comment:
This doesn't belong here, validating this data is `touchid.Register`'s responsibility. I would revert the file.